### PR TITLE
automate pypi releases

### PR DESCRIPTION
**Pull request by houqp**

### CODE REVIEW

File: rust/Cargo.toml [hunk 1]
```diff
@@ -30,9 +30,9 @@ azure_storage = { git = "https://github.com/Azure/azure-sdk-for-rust", optional
 rusoto_core = { version = "0.43", optional = true }
 rusoto_s3 = { version = "0.43", optional = true }
 
-arrow  = { git = "https://github.com/apache/arrow.git", branch = "master" }
-datafusion = { git = "https://github.com/apache/arrow.git", branch = "master", optional = true }
-parquet = { git = "https://github.com/apache/arrow.git", branch = "master" }
+arrow  = { git = "https://github.com/apache/arrow.git", rev = "6c3547347e9d95f7d0c77d5949cb8fcf6983ca9b" }
```
Comment (houqp) [inline rust/Cargo.toml:33]: Unfortunately we can't use branch constraint due to https://github.com/PyO3/maturin/issues/389

File: python/Cargo.toml [hunk 2]
```diff
@@ -24,12 +24,13 @@ version = "0.*"
...
 
 [package.metadata.maturin]
+name = "deltalake"
 classifier = [
     "Development Status :: 3 - Alpha",
     "License :: OSI Approved :: Apache Software License",
     "Programming Language :: Python :: 3 :: Only"
 ]
-project-url = ["Repo, https://github.com/delta-io/delta.rs"]
+project-url = { Repo = "https://github.com/delta-io/delta.rs" }
```
Comment (rtyler) [inline python/Cargo.toml:33]:
```suggestion
project-url = { Repo = "https://github.com/delta-io/delta-rs" }
```

File: .github/workflows/python_build.yml [hunk 3]
```diff
@@ -61,8 +58,9 @@ jobs:
     # - name: Run Rust tests
     #   run: cargo test
 
-    - name: 'Install Python build Dependencies'
-      run: pip install maturin
+    - name: Install matruin
```
Comment (rtyler) [inline .github/workflows/python_build.yml:61]:
```suggestion
    - name: Install maturin
```

File: .github/workflows/python_release.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
+name: Release
```
Comment (rtyler) [inline .github/workflows/python_release.yml:1]:
```suggestion
name: Release to PyPI
```

File: .github/workflows/python_release.yml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+          profile: minimal
+          toolchain: stable
+          override: true
+          target: ${{ matrix.target }}
+
+      - name: Install matruin
+        run: |
+          cargo install --git https://github.com/PyO3/maturin.git --rev 98636cea89c328b3eba4ebb548124f75c8018200 maturin
+
+      - uses: actions/setup-python@v2
```
Comment (rtyler) [inline .github/workflows/python_release.yml:57]: I'm assuming that by setting up all these Python point releases, a binary wheel will be pushed for each point release?

File: rust/Cargo.toml (hunk 1 above)
Comment (rtyler) [reply rust/Cargo.toml:33]:
```suggestion
# Cannot use branch constraint due to https://github.com/PyO3/maturin/issues/389
arrow  = { git = "https://github.com/apache/arrow.git", rev = "6c3547347e9d95f7d0c77d5949cb8fcf6983ca9b" }
```

File: .github/workflows/python_release.yml (hunk 5 above)
Comment (houqp) [reply .github/workflows/python_release.yml:57]: that's right, maturin will pick up all available python versions from PATH

**Comment by houqp** [comment]
here is what the artifact looks like in test pypi server: https://test.pypi.org/project/deltalake/0.1.0/#files

**Review by rtyler — APPROVED** [review]

**Pull request merged**